### PR TITLE
Make operations and error control added

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
Now, the calculator performs operations and controls errors

Operations and errors:

operation

Negative numbers:

negative numbers

### CODE REVIEW

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt [hunk 1]
```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
     }
 
     override fun getLastModified(): String {
-        return if (secondOperand.isNotEmpty()) {
-            secondOperand
-        } else if (operator.isNotEmpty()) {
-            operator
-        } else {
-            firstOperand
+        return when {
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt:52]: NIICE!

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt [hunk 2]
```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
...
+                operator
+            }
+            else -> {
+                firstOperand
+            }
+        }
+    }
+
+    private fun operandEnabled(operand: String): Boolean {
+        return (operand != EMPTY_STRING && operand != SUB)
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt]: here you can simplify more... like this 
private fun operandEnabled(operand: String) = operand != EMPTY_STRING && operand != SUB

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt [hunk 3]
```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
...
+            return Error.ERROR_INCOMPLETE_OPERATION
+        }
+        if (operandEnabled(firstOperand) && (operator == DIV) && (secondOperand == ZERO_STRING)) {
+            return Error.ERROR_DIVISION_BY_ZERO
+        }
+        return Error.NONE
+    }
+
+    private fun makeOperation(): Double {
+        when (operator) {
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt]: pls use here return when

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt [hunk 4]
```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
...
+            }
+            DIV -> {
+                return firstOperand.toDouble() / secondOperand.toDouble()
+            }
+        }
+        return DEFAULT_RESULT
+    }
+
+    override fun getOperation(): String {
+        return "$firstOperand $operator $secondOperand"
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt]: NICEE! same you can simplify 
override fun getOperation() = "$firstOperand $operator $secondOperand"

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt [hunk 5]
```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
...
+    private fun operationEnabled(): Boolean {
+        var operationEnabled = true
+        if (getError() == Error.ERROR_DIVISION_BY_ZERO || getError() == Error.ERROR_INCOMPLETE_OPERATION) {
+            operationEnabled = false
+        }
+        return operationEnabled
+    }
+
+    override fun getError(): Error {
+        if (!operandEnabled(firstOperand) || !operandEnabled(secondOperand)) {
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt:76]: here maybe you can use return if

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt (hunk 4 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt]: Solved!

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt (hunk 3 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt]: Solved!

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt (hunk 2 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt]: Solved!

File: app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt (hunk 5 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt:76]: You can't because they are two separate ifs

**Review by BQuinteros — APPROVED** [review]

**Review by LuchoElissondoGlobant — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Pull request merged**